Auth0/AuthProvider.swift

```diff
@@ -35,8 +35,27 @@ import UIKit
          ...
          return transaction
      }
+
  ```
  */
 public protocol AuthProvider {
     func login(withConnection connection: String, scope: String, parameters: [String: Any]) -> NativeAuthTransaction
+
+    /**
+     Some IdP authenticators like Auth0Twitter rely upon accessibility of on device social accounts.
+     If these are not accessible/available this pre-login check lets you know you should provide
+     alternative authentication means such as Auth0.WebAuth.
+     
+     - returns: Bool
+     */
+    func canUseNative() -> Bool
```

Comment:
why not something more common like `isAvailable()` or `available` prop both static. Also the comment is a bit misleading since we are dealing Auth providers so something like

```
Determine if the Auth method used by the Provider is available in the device. e.g. if using a Twitter Auth provider it should check the presence of a Twitter account in the device.

If a Auth is performed on a provider that returns `false` the transaction will fail with an error

- returns: Bool if the AuthProvider is available on the device
```